lib/watcher.js

```diff
@@ -267,14 +268,22 @@ class Watcher {
 				// the busy promise to fulfil
 				this.debouncer.cancel();
 				this.clearLogOnNextRun = false;
-				this.rerunAll();
+				if (data === 'u') {
+					this.rerunWithUpdate();
+				} else {
+					this.rerunAll();
+				}
 			});
 		});
 	}
 	rerunAll() {
 		this.dirtyStates = {};
 		this.run();
 	}
+	rerunWithUpdate() {
+		this.dirtyStates = {};
+		this.run(undefined, {updateSnapshots: true});
```

Comment:
Let's do that here as well then. Storing the [`specificFiles || files` argument here](https://github.com/efegurkan/ava/blob/08675e83375781fcf9468ce30d95e1e84f87f7d4/lib/watcher.js#L115) should be sufficient.